🚀 feat(barter): Implement Full Barter Lifecycle and Profile/Feed Integration at Backend

61021111 opened the pull request:
## Summary

This is a major feature PR that connects multiple core app services. It implements the complete end-to-end bartering flow, from initiation (via Search, Feed, or Wishlist) to the final accept/reject/notify cycle.

This PR also populates the new 'User Profile' tab with all relevant data (GPS, library, taste, counts) and links the Home Feed's review buttons (Like, Comment, Barter, Bookmark) to their respective backend services and the notification system.

## Changes Made

1.  **User Profile Tab Implementation:**
    *   Populated the user profile tab with dynamic data: GPS location, 'My Library', 'Wishlist', 'Reading Taste' (with skippable/follow-up logic), 'Barter Style', and counts (followers, following, reviews).

2.  **Home Feed Button Integration:**
    *   Connected the review card buttons (Like, Comment, Barter, Bookmark) to their respective backend services.
    *   All actions (Like, Comment, Barter Request) now log notifications for the recipient.
    *   The 'Bookmark' action is now synchronized with the user's main 'Wishlist'.

3.  **End-to-End Barter Flow:**
    *   Implemented the full barter lifecycle:
        *   Users can initiate a barter request from Search, Wishlist, or a Review card.
        *   Fetches book owners within a 20km radius (based on User GPS).
        *   Users can view detailed barter info before sending a request.
        *   Handles the complete Request, Accept, and Reject logic, including all related notifications.

4.  **Barter Availability Toggle:**
    *   Added an `is_for_barter` toggle for each user-owned book.
    *   Barter-inactive books are now correctly excluded from search/request candidates.

5.  **Payload Enhancements:**
    *   Created a new `UserBarterInfo` payload (includes library, wishlist, taste, style, distance, counts).
    *   Added `bookAvailableForBarter` field to the `PostSerializer`.

6.  **Database Migrations:**
    *   Added `latitude` and `longitude` fields to the `User` model.
    *   Added trade location, place, and coordinate fields to the `UserTaste` model.
    *   Included all necessary migration files.

## Testing

*   All local tests passed (`python manage.py test`).
*   Manually tested the full barter flow:
    *   User A (Request) -> User B (Receives Notification)
    *   User B (Accept/Reject) -> User A (Receives Notification)
*   Confirmed Home Feed buttons (Like, Comment, Barter, Bookmark) are correctly linked and functional.
*   Verified that user profile data is displayed correctly.

## Pre-Submission Checklist

### Code Quality
- [x] Code follows project coding conventions
- [x] Self-review completed
- [ ] Code is well-commented (especially complex logic)
- [x] No new warnings or errors (All `ruff` and `black` checks pass)

### Testing Requirements
- [x] Tests added/updated for new functionality
- [x] All tests pass locally
- [ ] Test coverage ≥80% for new code (N/A - will check in CI)
- [ ] Integration tests added (if applicable)

### Documentation
- [ ] Code documentation updated (docstrings, comments)
- [ ] API documentation updated (if API changes)
- [ ] README updated (if setup/usage changes)
- [x] Design Documentation updated on Wiki (if architectural changes) - *DB Migration*

### Dependencies & Breaking Changes
- [x] No new dependencies
- [x] No breaking changes

### Custom Checklist (from user)
- [x] Feature, Notification, and Payload systems are all connected
- [x] Database migration file is included
- [x] Rebased on `dev` and confirmed no merge conflicts

61021111 committed:
feat(profile,barter): Integrate User Profile Data and Barter Flow

61021111 committed:
fix(accounts): Exclude precise coordinates (PII) from UserBarterInfo payload

vnny27 commented on backend/barter/views.py:
```diff
@@ -1 +1,237 @@
+      - preferred_meeting_type: str (optional)
+      - proposed_meeting_location: str (optional)
+      - proposed_meeting_time: datetime (optional)
+    """
```
이 부분에서 requested_book_id가 어떤 의미인지 잘 모르겠어요. 내가 교환할 책인지 상대방의 책인지. 둘 다 필요할 거 같아서 필드 하나가 더 필요할 거 같아요. 또 프론트엔드의 홈 피드 post에서 바로 bartering으로 넘어가기 위해 post를 backend에서 건네 줄 때도 book_id가 포함되어서 건네주어야할 거 같아요.

61021111 replied:
제가 생각한 Barter의 로직은 다음과 같습니다!
1-1. 유저 A가 읽고싶은 책이 있어서 검색을 하거나 본인 위시리스트에 있는 책을 눌렀을때 거기에 교환하기 버튼이 있어서 그 버튼을 누르면 교환할 수 있는 사람 B,C,D (그 책을 가지고 있고, 그 책을 팔겠다고 설정해놨고, 내 주변 20km?에 있는 사람) 리스트가 나오는 화면으로 넘어간다. 그 화면에서 유저들 B, C, D를 하나씩 클릭해보면 그 유저 B, C, D의 정보가 나온다. 정보는 나와의 거리, 거래위치, 장소, 위시리스트, 서재, 서평 등등 거래에 도움이 되는 정보들을 포함하고 있다. (내 user tab 자체로 넘어가도 될 것 같습니다.) 그러면 그 유저들 중에서 한 유저 B를 골라서 교환하기 버튼을 누릅니다. 

1-2. 유저 A가 홈피드에 있는 서평 중에서 맘에드는 책을 발견해서 그 서평을 올린 유저 E에게 그 서평에 있는 교환버튼을 눌러서 바로 교환을 신청합니다.

2. 신청받은 유저 B or E에게 알림이 가서 그 알림을 클릭하면 A 유저의 정보들을 볼 수 있고, 거기에 역시 교환하기 혹은 거절하기 버튼이 있어서 결정을 내릴 수 있습니다.

제 barter 로직에 의하면 상대방에게 교환될 내 책을 고르는 과정이 없다보니 requested_book_id는 상대방이 가지고 있는 내가 원하는 책이었던 것 같습니다. 조금 더 생각해보자면, 교환 신청을 받은 사람이 교환하고 싶은 책을 고르는 과정이 필요할 수도 있을 것 같고 그게 아니라면 애초에 교환을 신청할때 본인 책을 골라서 교환을 신청할 수도 있을 것 같아요! 

아무래도 Barter 로직 관련해서는 팀원들끼리 논의를 해봐야할 것 같습니다. 

뒤에 말씀해주신 내용은 수정해서 다시 올리겠습니다. 감사합니다!

61021111 committed:
feat(social): add bookId to PostSerializer for direct bartering from home feed

Include related book's UUID in post serialization so frontend can initiate barter requests directly from home feed posts without additional lookups
